includes/embeds/class-amp-scribd-embed-handler.php

```diff
@@ -0,0 +1,93 @@
+<?php
+/**
+ * Class AMP_Scribd_Embed_Handler
+ *
+ * @package AMP
+ * @since 1.3.1
+ */
+
+/**
+ * Class AMP_Scribd_Embed_Handler
+ */
+class AMP_Scribd_Embed_Handler extends AMP_Base_Embed_Handler {
+
+	/**
+	 * Registers embed.
+	 */
+	public function register_embed() {
+		add_filter( 'embed_oembed_html', [ $this, 'filter_embed_oembed_html' ], 10, 2 );
+	}
+
+	/**
+	 * Unregisters embed.
+	 */
+	public function unregister_embed() {
+		remove_filter( 'embed_oembed_html', [ $this, 'filter_embed_oembed_html' ] );
+	}
+
+	/**
+	 * Filter oEmbed HTML for Scribd to be AMP compatible.
+	 *
+	 * @param string $cache Cache for oEmbed.
+	 * @param string $url   Embed URL.
+	 * @return string Embed.
+	 */
+	public function filter_embed_oembed_html( $cache, $url ) {
+		if ( ! in_array( wp_parse_url( $url, PHP_URL_HOST ), [ 'scribd.com', 'www.scribd.com' ], true ) ) {
+			return $cache;
+		}
+
+		$embed = $this->remove_script( $cache );
+		$embed = $this->inject_sandbox_permissions( $embed );
+
+		return $embed;
+	}
+
+	/**
+	 * Remove the 'script' element from the provided HTML if there is any.
+	 *
+	 * @param string $html HTML string.
+	 * @return string
+	 */
+	private function remove_script( $html ) {
+		$html_without_script = preg_replace( '#<script(?:\s.*?)?>.+?</script>#s', '', $html );
+
+		if ( null !== $html_without_script ) {
+			return $html_without_script;
+		}
+
+		return $html;
+	}
+
+	/**
+	 * Injects the 'allow-popups' & 'allow-scripts' permissions into the sandbox attribute so that
+	 * the 'Fullscreen' button works as intended.
+	 *
+	 * @param string $html HTML string.
+	 * @return string
+	 */
+	private function inject_sandbox_permissions( $html ) {
+		if ( preg_match( '#<iframe.+?sandbox="(?P<sandbox_attr>.+?)"#s', $html, $matches ) ) {
+			$permissions = [
+				'allow-popups',
+				'allow-scripts',
+			];
+
+			foreach ( $permissions as $permission ) {
+				if ( false === strpos( $matches['sandbox_attr'], $permission ) ) {
+					$matches['sandbox_attr'] .= " ${permission}";
+				}
+			}
+
+			return preg_replace(
+				'#(<iframe.+?sandbox=")(.+?)(".+</iframe>)#s',
+				"$1${matches['sandbox_attr']}$3",
+				$html,
+				1
+			);
+		} else {
+			return str_replace( '<iframe ', '<iframe sandbox="allow-scripts allow-popups" ', $html );
+		}
+	}
```

Comment:
I think this can be simplified a bit which will also eliminate the need for `remove_script`. 

Like this:

```php
private function sanitize_iframe( $html ) {
	return preg_replace_callback(
		'#^.*<iframe(?P<iframe_attributes>[^>]+?)></iframe>.*$#s',
		function ( $matches ) {
			$attr = $matches['iframe_attributes'];

			// Amend the required keywords to the iframe's sandbox.
			$sandbox  = 'allow-popups allow-scripts';
			$replaced = 0;
			$attr = preg_replace(
				'/(?<=\ssandbox=["\'])/',
				$sandbox,
				$attr,
				1,
				$replaced
			);

			// If no sandbox attribute was found, then add the attribute.
			if ( 0 === $replaced ) {
				$attr .= sprintf( ' sandbox="%s"', $sandbox );
			}

			// The iframe sanitizer will convert this into an amp-iframe.
			return "<iframe{$attr}></iframe>";
		},
		$html
	);
}
```

It's fine for `sandbox` keywords to be repeated.

Reply:
So simple, yet I never thought of something like that. I have much to learn from this project. Thanks @westonruter!